dateutil/parser/_parser.py

```diff
@@ -1115,16 +1115,18 @@ def _build_tzinfo(self, tzinfos, tzname, tzoffset):
             tzdata = tzinfos(tzname, tzoffset)
         else:
             tzdata = tzinfos.get(tzname)
+            # user opted to not pass in tzinfos
+            # or tzinfos is set to None  e.g { 'BRST' : None}
+            # don't punish them. 
+            if tzdata is None:
+                return
```

Comment:
I prefer to avoid multiple returns if possible. I would probably refactor this by changing line 1124 below to:

`if isinstance(tzdata, datetime.tzinfo) or tzdata is None:`

Reply:
+1